lasagne/updates.py

```diff
@@ -440,6 +440,12 @@ def rmsprop(loss_or_grads, params, learning_rate=1.0, rho=0.9, epsilon=1e-6):
            Neural Networks for Machine Learning, Lecture 6.5 - rmsprop.
            Coursera. http://www.youtube.com/watch?v=O3sxAc4hxZU (formula @5:20)
     """
+    # Converting input arguments to theano expressions so that
+    # float32 inputs arent upcasted to float 64 automatically
+    learning_rate = utils.as_theano_expression(learning_rate)
+    rho = utils.as_theano_expression(rho)
+    epsilon = utils.as_theano_expression(epsilon)
```

Comment:
I think I'd prefer only converting what's needed. Something like:
```python
# Convert rho to theano expression so (1 - rho) will never be casted to float64
rho = utils.as_theano_expression(rho)
```
The others don't risk upcasting anyway. Similar for adam and adamax: only beta1, beta2 need to be converted.